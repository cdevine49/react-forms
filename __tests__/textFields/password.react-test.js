import React from 'react';
import Password from '../../lib/components/textFields/password';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import 'jest-styled-components';

describe("Password", () => {
  describe("Default", () => {
    const valid = "Password1";
    const noNumbers = "Password";
    const noLowerCase = "PASSWORD1"
    const noUpperCase = "password1"
    const tooShort = "pswrd1"
    const tooLong = "Password1Password1Password1Password1Password1Password1Password1Password1Password1Password1Password1Password1Password1Password1Password1Password1"
    const pasword = shallow(
      <Password />
    );

    const errors = pasword.props().errors;

    test("9 letter password with uppercase, lowercase, and number is valid", () => {
      expect(errors.some(error => error._handle(valid))).toBe(false);
    });

    test("Must contain one or more numbers", () => {
      expect(errors.some(error => error._handle(noNumbers))).toBe(true);
    });

    test("Must contain one or more lowercase letters", () => {
      expect(errors.some(error => error._handle(noLowerCase))).toBe(true);
    });

    test("Must contain one or more uppercase letters", () => {
      expect(errors.some(error => error._handle(noUpperCase))).toBe(true);
    });

    test("Must be more than 8 characters long", () => {
      expect(errors.some(error => error._handle(tooShort))).toBe(true);
    });

    test("Must be less than 100 characters long", () => {
      expect(errors.some(error => error._handle(tooLong))).toBe(true);
    });

    test("has correct props", () => {
      const tree = renderer.create(
        <Password />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});