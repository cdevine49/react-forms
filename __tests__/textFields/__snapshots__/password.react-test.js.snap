// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password Default has correct props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  margin-bottom: 15px;
}

.c1 {
  color: #222;
  font-weight: bold;
  margin-bottom: 3px;
}

.c2 {
  background: #FFFFFF;
  line-height: 20px;
  vertical-align: middle;
  padding: 6px 8px;
  color: #222;
  vertical-align: middle;
  border: 1px solid #ddd;
  border-radius: 3px 3px;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.075);
  background-repeat: no-repeat;
  background-position: right 8px center;
  outline: none;
}

.c2:hover {
  border-color: #b9b9b9;
  border-top-color: #a0a0a0;
}

.c2:focus {
  border-color: #4d90fe;
}

<div
  className="c0"
>
  <label
    className="c1"
    htmlFor={undefined}
  >
    Password
  </label>
  <input
    aria-describedby={undefined}
    aria-invalid={false}
    aria-required={false}
    className="c2"
    id={undefined}
    onBlur={[Function]}
    onChange={[Function]}
    type="password"
    value=""
  />
</div>
`;
